src/EventStore.Core/Services/Transport/Grpc/PersistentSubscriptions.Read.cs

```diff
@@ -238,41 +242,29 @@ private class PersistentStreamSubscriptionEnumerator
 				}
 
 				void Fail(Exception ex) {
-					_sendQueue.Enqueue((default, default, ex));
+					_channel.Writer.TryComplete(ex);
 					_subscriptionIdSource.TrySetException(ex);
 				}
-
-				void EnqueueEvent(ResolvedEvent resolvedEvent, int retryCount) =>
-					_sendQueue.Enqueue((resolvedEvent, retryCount, null));
 			}
 
 			public ValueTask DisposeAsync() {
 				_publisher.Publish(new ClientMessage.UnsubscribeFromStream(Guid.NewGuid(), _correlationId,
 					new NoopEnvelope(), _user));
-				_disposedTokenSource.Dispose();
-				return _tokenRegistration.DisposeAsync();
+				return new ValueTask(Task.CompletedTask);
```

Comment:
@thefringeninja It seems a call to `_channel.TryComplete()` is missing before this line, or is it intentional?